### ISSUE TITLE
OMPL source flattens first level of folders and ignores feeds in deeply nested folders

### ISSUE THREAD

**Issue by liskin** [lwindolf/liferea#840]
When the following OPML is added as a source:

```xml
<?xml version="1.0"?>
<opml version="1.0">
  <head>
    <title>title0</title>
  </head>
  <body>
    <outline text="top feed" type="atom" xmlUrl="file:///tmp/feed1.xml" />
    <outline text="top folder">
      <outline text="nested feed" type="atom" xmlUrl="file:///tmp/feed2.xml" />
      <outline text="nested folder" >
      </outline>
    </outline>
  </body>
</opml>
```

It results in this:

![2020-06-11-152124_275x158_scrot](https://user-images.githubusercontent.com/300342/84390089-5e902c00-abf7-11ea-96e9-0309ab6f1ad4.png)

It it's then modified:

```xml
<?xml version="1.0"?>
<opml version="1.0">
  <head>
    <title>title0</title>
  </head>
  <body>
    <outline text="top feed" type="atom" xmlUrl="file:///tmp/feed1.xml" />
    <outline text="top folder">
      <outline text="nested feed" type="atom" xmlUrl="file:///tmp/feed2.xml" />
      <outline text="nested folder" >
        <outline text="deep nested feed" type="atom" xmlUrl="file:///tmp/feed3.xml" />
      </outline>
    </outline>
  </body>
</opml>
```

and refreshed, `opml_source_merge_feed(): bad! bad! very bad!` is logged and the "deep nested feed" is not added at all.

I believe this is because https://github.com/lwindolf/liferea/blob/df84f2118564b479b917e7a0c264a7cb47fd4932/src/fl_sources/opml_source.c#L80 sets parent to `mergeCtxt->rootNode` instead of `mergeCtxt->parent`.

**Comment by lwindolf** [comment]
Thank you for reporting this *and* suggesting the fix!!!

**Comment by lwindolf** [comment]
Fix released with 1.12.8
